### PR TITLE
BANNO.ODTOPTIN.V1.POW add OTA default

### PR DESCRIPTION
**Pull request by tkainz**
Added the ability to set default OT values for share types not specifically assigned an OT amount value.

### CODE REVIEW

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 1]
```diff
@@ -949,7 +987,52 @@ PROCEDURE READCONFIGFILESETTINGS
...
+       IF PARAMOTASHARETYPE(OTALOOP)=TMPLOOP THEN
+        DO
+         SHARETYPEFOUND=TRUE
+         OTAAMOUNT=PARAMOTAAMOUNT(OTALOOP)
+        END
+      END [FOR OTALOOP=1 TO PARAMOTACOUNT]
+
+     BNODEBUGMSG=FORMAT("Evaluating OTA parm list for share type 9999",TMPLOOP)+
+                 "  SHARE TYPE FOUND="+BNOTRUEFALSECHR(SHARETYPEFOUND)
+
```
Comment (jucarson) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:1012]: It appears you are missing a call to the BNODEBUGMSGADD procedure. I did test, and this "Evaluating OTA..." line is never sent to the output. Also, in that same line, I think you have misspelled param (parm)

**Review by jucarson — CHANGES_REQUESTED** [review]
Just one comment/change requested, otherwise, looks good.

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW (hunk 1 above)
Comment (tkainz) [reply odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:1013]: @jucarson Corrections made - thanks!

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 2]
```diff
@@ -866,11 +895,20 @@ PROCEDURE READCONFIGFILESETTINGS
...
+     CALL BNODEBUGMSGADD
+    END
    FOR TMPLOOP=1 TO PARAMOTACOUNT
     DO
-     BNODEBUGMSG=FORMAT("OTA 999 -",TMPLOOP)+
-                 FORMAT(" SHARE TYPE 9999  ",PARAMOTASHARETYPE(TMPLOOP))+
-                 FORMAT("OT AMOUNT $###,###,##9.99",PARAMOTAAMOUNT(TMPLOOP))
+     TMPCHR=FORMAT("###,###,##9.99",PARAMOTAAMOUNT(TMPLOOP))
+     CALL NLTS
+     BNODEBUGMSG=FORMAT("  OTA 999 -",TMPLOOP)+
```
Comment (jordan-keenan) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:909]: I suggest changing "OTA 999 -" in the configs debug to "OTA 99 -" as 99 is the max

File: odt-opt-in/LETTERSPECS/BANNO.ODTOPTIN.V1.CFG [hunk 3]
```diff
@@ -122,12 +124,19 @@ AFF:7
 *
 * OVERDRAFT TOLERANCE AMOUNT
 *  Line must begin with "OTA"
-*  Overdraft Tolerance amount by share type.  When opting-in, this amount will be set in the Overdraw Tolerance field
-*  in the share record.  The amount is mapped based on the SHARE:TYPE field.
+*  Overdraft Tolerance amount by share type. When opting-in, this amount will be set in the Overdraw Tolerance field
```
Comment (jordan-keenan) [inline odt-opt-in/LETTERSPECS/BANNO.ODTOPTIN.V1.CFG:127]: I suggest stating in the comments that a maximum of 99 OTA definitions are allowed

**Review by jucarson — APPROVED** [review]
Looks Good. Thanks Tom!

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW (hunk 2 above)
Comment (tkainz) [reply odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW]: @jordan-keenan Done

File: odt-opt-in/LETTERSPECS/BANNO.ODTOPTIN.V1.CFG (hunk 3 above)
Comment (tkainz) [reply odt-opt-in/LETTERSPECS/BANNO.ODTOPTIN.V1.CFG:127]: @jordan-keenan Done

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 4]
```diff
@@ -949,7 +987,53 @@ PROCEDURE READCONFIGFILESETTINGS
...
+
+     BNODEBUGMSG=FORMAT("Evaluating OTA param list for share type 9999",TMPLOOP)+
+                 "  SHARE TYPE FOUND="+BNOTRUEFALSECHR(SHARETYPEFOUND)
+     CALL BNODEBUGMSGADD
+
+     IF SHARETYPEFOUND=FALSE AND
+        DEFAULTOTAVALUE>=$0.00 THEN
+      OTAAMOUNT=DEFAULTOTAVALUE
+
+     PARAMOTAAMOUNT(OTALOOP)=OTAAMOUNT
```
Comment (jordan-keenan) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:1018]: I don't this this line 1018 would work properly as it is using OTALOOP as the iterator; but the OTA Loop is closed at this point in the code.
Or am I missing something?

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 5]
```diff
@@ -949,7 +987,53 @@ PROCEDURE READCONFIGFILESETTINGS
...
+
+     TMPCHR=FORMAT("###,###,##9.99",OTAAMOUNT)
+     CALL NLS
+     OTAAMOUNTCHR=TMPCHR
+
+     BNODEBUGMSG=FORMAT("VALID SHARE TYPE: 9999 - OTA PARAM FOUND: ",TMPLOOP)+
+                 BNOTRUEFALSECHR(SHARETYPEFOUND)
+     IF SHARETYPEFOUND=TRUE THEN
+      BNODEBUGMSG=BNODEBUGMSG+" FOR: $"+OTAAMOUNTCHR+".  "+
```
Comment (jordan-keenan) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:1027]: @tkainz For line 1027 OTAAMOUNTCHR is printed, then the same variable, OTAAMOUNTCHR,  is printed again in line 1028, both as part of the same debug line. Did you mean to use a different variable for one of them?

**Review by jordan-keenan — APPROVED** [review]
Code changes look good, @tkainz !

**Review by jucarson — APPROVED** [review]
Recent changes look good.

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 6]
```diff
@@ -687,14 +701,29 @@ PROCEDURE READCONFIGFILESETTINGS
...
+           IF UPPERCASE(SEGMENT(PARAMVALUE,1,COLONPOS-1))="XXXX" THEN
+            DO
+             DEFAULTOTAVALUE=MONEY(VALUE(SEGMENT(PARAMVALUE,COLONPOS+1,LENGTH(PARAMVALUE))))
+             TMPCHR=FORMAT("###,###,##9.99",DEFAULTOTAVALUE)
+             CALL NLS
+             DEFAULTOTAVALUECHR=TMPCHR
+             BNODEBUGMSG="Default OTA value FOUND: $"+DEFAULTOTAVALUECHR
+             CALL BNODEBUGMSGADD
+            END
+           ELSE
```
Comment (rrobison44) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:715]: Could just do:
```
ELSE IF PARAMOTACOUNT<PARAMOTAMAX THEN
```
to get rid of a couple lines of code.

**Review by rrobison44 — APPROVED** [review]
Added a couple comments.

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW (hunk 6 above)
Comment (tkainz) [reply odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:715]: @rrobison44 changed.

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 7]
```diff
@@ -31,6 +31,8 @@
 **                         configuration file validation error.
 **    Ver. 1.2.0  05/14/24 TKainz - Banno
 **                         Added Program Installation Date parameter
+**    Ver. 1.2.1  08/02/24 TKainz - Banno
```
[the remark this answers is not in the document]
Comment (tkainz) [reply odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:34]: @rrobison44 updated as feature

File: odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW [hunk 8]
```diff
@@ -194,18 +196,29 @@ DEFINE
  PARAMINSTALLDATE             = DATE
  PARAMDEBUGMODE               = NUMBER
  PARAMDEBUGMODEFOUND          = NUMBER
+
+[* Added 08/02/24 *]
```
Comment (jordan-keenan) [inline odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW:200]: I think this (line 200) should match the PR date (in this case it was 08/13/24)

**Pull request merged**